frigate/events.py

```diff
@@ -85,81 +82,82 @@ def run(self) -> None:
+                first_detector = list(self.config.detectors.values())[0]
+
+                start_time = event_data["start_time"] - event_config.pre_capture
+                end_time = (
+                    None
+                    if event_data["end_time"] is None
+                    else event_data["end_time"] + event_config.post_capture
+                )
+                # score of the snapshot
+                score = (
+                    None
+                    if event_data["snapshot"] is None
+                    else event_data["snapshot"]["score"]
+                )
+                # detection region in the snapshot
+                region = (
+                    None
+                    if event_data["snapshot"] is None
+                    else to_relative_box(
+                        width,
+                        height,
+                        event_data["snapshot"]["region"],
+                    )
+                )
+                # bounding box for the snapshot
+                box = (
+                    None
+                    if event_data["snapshot"] is None
+                    else to_relative_box(
+                        width,
+                        height,
+                        event_data["snapshot"]["box"],
+                    )
+                )
+
+                # keep these from being set back to false because the event
+                # may have started while recordings and snapshots were enabled
+                # this would be an issue for long running events
+                if self.events_in_process[event_data["id"]]["has_clip"]:
+                    event_data["has_clip"] = True
+                if self.events_in_process[event_data["id"]]["has_snapshot"]:
+                    event_data["has_snapshot"] = True
+
+                Event.replace(
                     id=event_data["id"],
                     label=event_data["label"],
                     camera=camera,
-                    start_time=event_data["start_time"] - event_config.pre_capture,
-                    end_time=None,
+                    start_time=start_time,
+                    end_time=end_time,
                     top_score=event_data["top_score"],
-                    false_positive=event_data["false_positive"],
+                    score=score,
                     zones=list(event_data["entered_zones"]),
                     thumbnail=event_data["thumbnail"],
-                    region=event_data["region"],
-                    box=event_data["box"],
-                    area=event_data["area"],
+                    region=region,
+                    box=box,
                     has_clip=event_data["has_clip"],
                     has_snapshot=event_data["has_snapshot"],
+                    model_hash=first_detector.model.model_hash,
+                    model_type=first_detector.model.model_type,
+                    detector_type=first_detector.type,
                 ).execute()
```

Comment:
I think this means that in-progress events that have `sub_label` or `retain_indefinitely` will have their values erased when this is called

Reply:
maybe there is some way to keep track if the event has been inserted or not, and call replace or update based on that?

Reply:
https://stackoverflow.com/a/33819556 seems to have some good options